templates/terraform/expand_property_method.erb

```diff
@@ -160,4 +155,12 @@ func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d T
   // TODO: Property '<%= property.name -%>' of type <%= property.class -%> is not supported
 }
 <%   end # tf_types.include?(property.class) -%>
+
+<% if property.nested_properties? -%>
+<%   property.nested_properties.each do |prop| -%>
+<%     next if property.is_a?(Api::Type::Map) && prop.name == property.key_name -%>
```

Comment:
Not that it wasn't confusing in the original location, but it's unintuitive why we care if the parent property is a `Map`. Do you mind explaining with a comment?

Reply:
The reason it's here is because `key_name` is only a property of Map's. So the `if Map` is just a guard and I'm not sure that's a great comment. Unfortunately I have no clue why we skip Map properties where the property.name == parent.key_name, which I think is the most actually helpful comment here :(

Reply:
Riley helped me out with the real explanation. Will be adding
"Map is a map from {key -> object} in the API, but Terraform can't represent that
so we treat the key as a property of the object but it isn't present as a property in the response" to the next PR